### PR TITLE
WIP: Improve docs for k8s.io/client-go/util/workqueue

### PR DESCRIPTION
**Pull request by pmorie**
Improves the godoc in package `k8s.io/client-go/util/workqueue`, correcting various typos and adding godoc to exported types.

/sig api-machinery

### CODE REVIEW

File: staging/src/k8s.io/client-go/util/workqueue/rate_limitting_queue.go [hunk 1]
```diff
@@ -16,46 +16,53 @@ limitations under the License.
...
+// failures forever.
 func NewRateLimitingQueue(rateLimiter RateLimiter) RateLimitingInterface {
 	return &rateLimitingType{
 		DelayingInterface: NewDelayingQueue(),
 		rateLimiter:       rateLimiter,
 	}
 }
 
+// NewNamedRateLimitingQueue constructs a new workqueue with rateLimited
+// queuing ability. Remember to call Forget!  If you don't, you may end up
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/util/workqueue/rate_limitting_queue.go]: nit: remove the extra space.

File: staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go [hunk 2]
```diff
@@ -27,15 +27,18 @@ import (
 type RateLimiter interface {
 	// When gets an item and gets to decide how long that item should wait
 	When(item interface{}) time.Duration
-	// Forget indicates that an item is finished being retried.  Doesn't matter whether its for perm failing
-	// or for success, we'll stop tracking it
+	// Forget indicates that an item is finished being retried.  Forget should
```
Comment (caesarxuchao) [inline staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go]: nit: remove the extra space.

**Review by caesarxuchao — COMMENTED** [review]
Changes look good to me. @deads2k if you want to take a look.

@pmorie is it still WIP?

**Comment by pmorie** [comment]
@caesarxuchao yeah, still WIP for now.  There are some comments I would like to expand significantly.

**Pull request closed without merging**